src/mono/mono/mini/mini-llvm-cpp.cpp

```diff
@@ -229,6 +229,13 @@ mono_llvm_build_weighted_branch (LLVMBuilderRef builder, LLVMValueRef cond, LLVM
 	return wrap (ins);
 }
 
+LLVMValueRef
```

Comment:
Why not just `LLVMBuildAShr`?

Reply:
I have no reason handy other than "mirroring clang's generated IR as closely as is reasonable". Non-exact ashr and non-nsw mul both generate a pmuldq in a small test so maybe this isn't necessary.